samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/transition.c

```diff
@@ -0,0 +1,456 @@
+
+			lightness_transition = 0;
+			k_timer_stop(&lightness_transition_timer);
+		}
+		break;
+	case LIGTHT_LIGHTNESS_ACTUAL_TT:
+		if (light_lightness_srv_user_data.tt_counter != 0) {
+
+			light_lightness_srv_user_data.tt_counter--;
+
+			tmp_u32 = light_lightness_srv_user_data.actual +
+				light_lightness_srv_user_data.tt_delta;
+
+			if (light_lightness_srv_user_data.actual != tmp_u32) {
+
+				light_lightness_srv_user_data.actual = tmp_u32;
+
+				state_binding(ACTUAL, IGNORE_TEMP);
+				update_light_state();
+			}
+		}
+
+		if (light_lightness_srv_user_data.tt_counter == 0) {
+			light_lightness_srv_user_data.actual =
+				light_lightness_srv_user_data.target_actual;
+
+			state_binding(ACTUAL, IGNORE_TEMP);
+			update_light_state();
+
+			lightness_transition = 0;
+			k_timer_stop(&lightness_transition_timer);
+		}
+		break;
+	case LIGTHT_LIGHTNESS_LINEAR_TT:
+		if (light_lightness_srv_user_data.tt_counter != 0) {
+
+			light_lightness_srv_user_data.tt_counter--;
+
+			tmp_u32 = light_lightness_srv_user_data.linear +
+				light_lightness_srv_user_data.tt_delta;
+
+			if (light_lightness_srv_user_data.linear != tmp_u32) {
+
+				light_lightness_srv_user_data.linear = tmp_u32;
+
+				state_binding(LINEAR, IGNORE_TEMP);
+				update_light_state();
+			}
+		}
+
+		if (light_lightness_srv_user_data.tt_counter == 0) {
+			light_lightness_srv_user_data.linear =
+				light_lightness_srv_user_data.target_linear;
+
+			state_binding(LINEAR, IGNORE_TEMP);
+			update_light_state();
+
+			lightness_transition = 0;
+			k_timer_stop(&lightness_transition_timer);
+		}
+		break;
+	case LIGTHT_CTL_TT:
+		if (light_ctl_srv_user_data.tt_counter != 0) {
+
+			light_ctl_srv_user_data.tt_counter--;
+
+			/* Lightness */
+			tmp_u32 = light_ctl_srv_user_data.lightness +
+				light_ctl_srv_user_data.tt_lightness_delta;
+
+			light_ctl_srv_user_data.lightness = tmp_u32;
+
+			/* Temperature */
+			tmp_u32 = light_ctl_srv_user_data.temp +
+				light_ctl_srv_user_data.tt_temp_delta;
+
+			light_ctl_srv_user_data.temp = tmp_u32;
+
+			/* Delta_UV */
+			tmp_s32 = light_ctl_srv_user_data.delta_uv +
+				light_ctl_srv_user_data.tt_delta_uv_delta;
+
+			if (tmp_s32 > 32767) {
+				tmp_s32 = 32767;
+			} else if (tmp_s32 < -32768) {
+				tmp_s32 = -32768;
+			}
+
+			light_ctl_srv_user_data.delta_uv = tmp_s32;
+
+			state_binding(CTL, CTL_TEMP);
+			update_light_state();
+		}
+
+		if (light_ctl_srv_user_data.tt_counter == 0) {
+			light_ctl_srv_user_data.lightness =
+				light_ctl_srv_user_data.target_lightness;
+
+			light_ctl_srv_user_data.temp =
+				light_ctl_srv_user_data.target_temp;
+
+			light_ctl_srv_user_data.delta_uv =
+				light_ctl_srv_user_data.target_delta_uv;
+
+			state_binding(CTL, CTL_TEMP);
+			update_light_state();
+
+			lightness_transition = 0;
+			k_timer_stop(&lightness_transition_timer);
+		}
+		break;
+	default:
+		break;
+	}
+}
+
+void temperature_transition_timer_handler(struct k_timer *dummy)
+{
+	s32_t tmp_s32;
+	u32_t tmp_u32;
```

Comment:
Variable declarations should be in the smallest possible scope, so please move them there. Also rename them more sensibly. It seems like `new_level` and `new_temp` would be much more descriptive.

Reply:
I will rename those. But if moved it to inner scope then unnecessarily I've too declare them twice.. will it do ?